web-app/src/app/store/saga/dataset-saga.ts

```diff
@@ -1,22 +1,27 @@
-import { type StrictEffect, call, takeLatest, put } from 'redux-saga/effects';
+import { call, takeLatest, put } from 'redux-saga/effects';
 import { loadingFeedFail } from '../feed-reducer';
 import { getAppError } from '../../utils/error';
 import { DATASET_LOADING_FEED, type FeedError } from '../../types';
 import { type PayloadAction } from '@reduxjs/toolkit';
 import { getGtfsFeedDatasets } from '../../services/feeds';
 import { type paths } from '../../services/feeds/types';
 import { loadingDatasetSuccess } from '../dataset-reducer';
+import { getUserAccessToken } from '../../services';
 
 function* getDatasetSaga({
-  payload: { feedId, accessToken },
-}: PayloadAction<{ feedId: string; accessToken: string }>): Generator<
-  StrictEffect,
-  void,
-  paths['/v1/gtfs_feeds/{id}/datasets']['get']['responses'][200]['content']['application/json']
-> {
+  payload: { feedId },
+}: PayloadAction<{ feedId: string }>): Generator {
   try {
     if (feedId !== undefined) {
-      const datasets = yield call(getGtfsFeedDatasets, feedId, accessToken, {});
+      const accessToken: string | null = (yield call(
+        getUserAccessToken,
+      )) as string;
```

Comment:
we don't need to declare `string | null` and `as string`

Reply:
Good observation. However, it's actually needed at list the `as string`. The reason is that generators functions don't play nice with TS with multiple calls. The root cause is that the generators functions "always" returns any. we were enforcing the type discovery defined  StriictEffect type but this doesn't account for multiple calls. The way I was able to go around is to define the return type a generic `Generator` vs the previous pattern was like `Generator<StrictEffect, void,AllFeedType>`.  
In any case, you are right I can skip the `string | null` definition keeping the `as string`, resulting code `const accessToken = (yield call(getUserAccessToken)) as string;`